sotodlib/io/hkdb.py

```diff
@@ -0,0 +1,425 @@
+    if isinstance(cfg, str):
+        cfg = HkConfig.from_yaml(cfg)
+    hkdb = HkDb(cfg)
+    session = hkdb.Session()
+    update_file_index(cfg, session=session)
+    update_frame_index(cfg, session=session)
+
+
+#####################
+# HK Loading stuff
+#####################
+@dataclass
+class Field:
+    agent: str
+    feed: str
+    field: str
+
+    def __str__(self):
+        return f"{self.agent}.{self.feed}.{self.field}"
+    
+    def matches(self, other):
+        if self.agent != other.agent:
+            return False
+        if self.feed != other.feed and self.feed != '*' and other.feed != '*':
+            return False
+        if self.field != other.field and self.field != '*' and other.field != '*':
+            return False
+        return True
+    
+    @classmethod
+    def from_str(cls, s):
+        try:
+            agent, feed, field = s.split('.')
+        except Exception:
+            raise ValueError(f"Could not parse field: {s}")
+
+        return cls(agent, feed, field)
+
+@dataclass
+class LoadSpec:
+    """
+    HK loading specification
+    """
+    cfg: HkConfig
+
+    fields: List[str]
+    """
+    List of field specifications to load. This can either be a field
+    descriptor, of the format ``agent.feed.field``, or an alias defined in
+    the config. Field descriptors can contain wildcards, for instance
+    ``agent.*.*`` will load all fields belonging to the specified agent.
+    ``agent.feed.*`` and ``agent.*.field`` will also work as expected.
+    """
+
+    start: float
+    "Start time to load"
+
+    end: float
+    "End time to load"
+
+    downsample_factor: int = 1
+    "Downsample factor for data"
+
+
+    def __post_init__(self):
+        fs = []
+        for f in self.fields:
+            if f in self.cfg.aliases:
+                fs.append(Field.from_str(self.cfg.aliases[f]))
+            else:
+                fs.append(Field.from_str(f))
+        self.fields = fs
+
+
+class HkResult:
+    """
+    Helper class for storing results of LoadHk. If aliases are set for any
+    of the keys, they will be set as attributes of the object.
+
+    Attributes
+    ------------
+    data: dict
+        Dict where the key is the field descriptor, and the value is a list
+        where val[0] are timestamps, and val[1] is data.
+    """
+    def __init__(self, data, aliases=None):
+        if aliases is None:
+            aliases = {}
+        self._aliases = aliases
+        self.data = data
+        for alias, key in aliases.items():
+            if key in self.data:
+                setattr(self, alias, self.data[key])
+
+    def save(self, path):
+        np.savez(path, aliases=np.array(self._aliases), **self.data)
+
+    @classmethod
+    def load(cls, path):
+        d = np.load(path, allow_pickle=True)
+        aliases = d['aliases'].item()
+        data = {k: d[k] for k in d.files if k != 'aliases'}
+        return cls(data, aliases=aliases)
+
+
+def load_hk(load_spec: LoadSpec, show_pb=False):
```

Comment:
Since this is a main entry point, it might be helpful to also accept "load_spec" as a dict that gets auto-converted to LoadSpec.  Also consider engine=... arg to pass through to HkDb.